src/inference/inference.ts

```diff
@@ -140,42 +140,50 @@ export class Inference {
           ' documents. See https://docs.pinecone.io/models for supported models.'
       );
     }
-    // Destructure `options` with defaults
-    // Note: If the user passes in key:value pairs in `options` that are not the following, they are ignored
+
     const {
       topN = documents.length,
       returnDocuments = true,
-      rankFields = ['text'],
       parameters = {},
+      rankFields = undefined,
     } = options;
 
-    // Allow documents to be passed a list of strings, or a list of objs w/at least a `text` key:
-    let newDocuments: Array<{ [key: string]: string }> = [];
-    if (typeof documents[0] === 'object' && !('text' in documents[0])) {
-      throw new PineconeArgumentError(
-        '`documents` can only be a list of strings or a list of objects with at least a `text` key, followed by a' +
-          ' string value'
-      );
-    } else if (typeof documents[0] === 'string') {
-      newDocuments = documents.map((doc) => {
-        return { text: doc as string };
-      });
+    let computedRankFields: string[];
+
+    // If user does not pass in rankFields:
+    if (!rankFields) {
+      // If `documents` is an array of strings:
+      if (documents.every((doc) => typeof doc === 'string')) {
+        // Set rankFields to default value of 'text'
+        computedRankFields = ['text'];
+      } else if (
+        // If`documents` is an array of objects:
+        documents.every((doc) => typeof doc === 'object' && doc !== null)
+      ) {
+        // If there is a `text` key (other keys can be present too):
+        if (
+          documents.every((doc) => (doc as { [key: string]: string })['text'])
+        ) {
+          computedRankFields = ['text'];
+        } else {
+          // If there is no `text` key:
+          throw new PineconeArgumentError(
+            'One or more documents are missing the specified rank field: `text`'
+          );
+        }
+      } else {
+        throw new PineconeArgumentError(
+          'Documents must be a list of strings or a list of objects'
+        );
+      }
```

Comment:
Is it possible for it to be a mix of the two? I'm actually not sure, but I'm assuming we don't allow that.

Reply:
Ah that's a good q! Right now, the TS client doesn't account for that. I'll add it in, thank you!

Reply:
Do you mean if someone mixed strings and documents in the same request? That would be kinda odd, I don't think we need to bend over backwards to support that

Reply:
okay cool. I think my final `else` stmt would actually account for this anyways, no?

The logic is:
- if docs is an array of strings, do something
- else if docs is an array of objs, do something else
- else, error out

Reply:
Yeah, I think the logic should be fine for that atm.